source/logrepl/iterator.go

```diff
@@ -138,7 +184,7 @@ func (i *CDCIterator) Teardown(ctx context.Context) error {
 
 // attachSubscription determines the starting LSN and key column of the source
 // table and prepares a subscription.
-func (i *CDCIterator) attachSubscription(ctx context.Context, conn *pgx.Conn) error {
+func (i *Iterator) attachSubscription(ctx context.Context, conn *pgx.Conn) error {
```

Comment:
I considered this exact case but I arrived at a different conclusion. My thought was that if a pipeline was stopped in the middle of operation, we would want to record the progress and thus the position, and that the manual intervention required would be necessary either way if they wanted to capture the snapshot. 

This is probably a good behavior to note in a connector development standards document somewhere.

Reply:
Sure manual intervention will be needed, but let's define what "manual intervention"  means. Right now that would mean recreating the connector, otherwise the pipeline won't start, which is a bit much IMO.

I would say we should rather tell the user to check what records were written on the destination and clean it up (i.e. remove them if needed), because the connector will produce the same records again (or similar ones, in case the table content changed in the meantime). Conduit advertises "at least once delivery" so duplicated messages can already happen and destinations should ideally implement idempotent writes, so this behavior wouldn't violate any expectations.

TL;DR: I think the connector should behave the same as the S3 connector and, in case the position passed to `Open` is a snapshot position just restart the snapshot from scratch and print a warning that we can't restart an interrupted snapshot.